modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpansionModuleModel.java

```diff
@@ -137,6 +140,10 @@ public ArtifactAst expand() {
   }
 
   private ArtifactAst expand(List<ComponentAst> moduleComponentModels, Set<String> moduleGlobalElementsNames) {
+    if (existOperationThatUsesImplicitConfiguration() && hasXmlSdkPropertiesWithDefaultValues()) {
+      applicationModel.topLevelComponents().add(new XmlSdkImplicitConfig(extensionModel));
```

Comment:
this should be handled by the copy below, not added directly.

Reply:
I haven't found a way to do that, I need the config first so the operations can reference it if they need it. Do you have an example?

Reply:
check `MacroExpansionModuleModel#expand(List<ComponentAst> moduleComponentModels, Set<String> moduleGlobalElementsNames)`, it passes `copyRecursively` a 3rd parameter that is a supplier of the top level elements to add. A copy method must be used intead of modifying the AST (which should be immutable , so the same concern as about the origin of that collection in the comment below applies here as well)